Ds 504

Fennersteel opened the pull request.

Fennersteel committed:
Began conversion of script from bash to python

Fennersteel committed:
Added env info

Fennersteel committed:
WIP

Fennersteel committed:
Updated test_irods_unixfilesystem_resource to verify that resource and all attributes were properly instantiated

Fennersteel committed:
wip

Fennersteel committed:
wip

tedgin committed:
Merge branch 'main' into DS-504

Fennersteel committed:
Finished conversion of irods_unix_filesystem to python 3.

Fennersteel committed:
Merge branch 'DS-504' of https://github.com/cyverse/ds-playbooks into DS-504

Fennersteel committed:
Fixed pylint conflicts

Fennersteel committed:
wip

Fennersteel committed:
Updated irods_unixfilesystem_resource and test play

- irods_unixfilesystem_resource
	- Updated "free_space" variable to be an integer of how much space the user would like allocated to the resource"

- test irods_unixfilesystem_resource.yml
	- Updated tests to verify resources and all attributes were properly instantiated

Commit:
Merge branch 'main' into DS-504

Fennersteel committed:
Rebase error fixes

Commit:
HOTFIX: make proxy rsyslog conf path configurable

Commit:
HOTFIX: reload haproxy when rsyslog restarted

tedgin reviewed (dismissed):
The idempotency check is failing. Please fix.

```console
? testing/test_module irods/library/ irods_unixfilesystem_resource
...
checking idempotency
...
TASK [create test irods environment] *******************************************
/usr/local/lib/python3.6/site-packages/ansible/parsing/vault/__init__.py:44: CryptographyDeprecationWarning: Python 3.6 is no longer supported by the Python core team. Therefore, support for it is deprecated in cryptography and will be removed in a future release.
  from cryptography.exceptions import InvalidSignature
changed: [localhost]

TASK [run script without optional vars] ****************************************
fatal: [localhost]: FAILED! => {"changed": false, "exc": "TypeError", "exc_message": "Unable to instantiate resource.", "exc_msg": "Value of unknown type: <class 'i********.exception.CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME'>, None", "msg": "Unable to run module.\nValue of unknown type: <class 'i********.exception.CATALOG_ALREADY_HAS_ITEM_BY_THAT_NAME'>, None", "response": ""}
...
```

Fennersteel committed:
Added functionality to retain idempotency
	- irods_unixfilesystem_up:
		- Added a check to the create_unix_resource to check if a resource of the given name
			has already been instantiated, and exit the program if so

tedgin committed:
HOTFIX: corrected HAProxy unix bind path

tedgin committed:
HOTFIX: wired rsyslog location through to role

Commit:
HOTFIX: workaround for bug in yum_versionlock

See https://github.com/ansible-collections/community.general/issues/4470
for details about the bug.  Previously, holding back the version of the
community.general collection to version 3.2 prevented the bug from
impacting us, but now this version can no longer be installed. Since the
bug only impacts idempotency, we decided to switch to the most recent
version of community.general and mark all yum_versionlock tasks as
non_idempotent.

Commit:
HOTFIX: fixed linting issues

tedgin committed:
HOTFIX: correct haproxy_irods.sock loc and nbthread val

The paths in bind and server references to haproxy_irods.sock no longer
rely on unix-bind prefix. This allows the master CLI socket to be
created in the folder /run.

nbthread is set to two less than the number of VCPUs with each thread
being bound to its own VCPU. This allows two threads to be used by the
OS.

tedgin committed:
HOTFIX: Installation instructions

tedgin committed:
Merge branch 'main' into DS-504

tedgin committed:
DS-504: Improved testing output

tedgin commented:
The idempotency check is still failing.  Please look at the output of the test harness to make sure everything passes before making a PR

Fennersteel committed:
Added steps in Docker setup to fix malfunction

Fennersteel committed:
Idempotency and resource attribute assignment changes
	- irods_unixfilesystem_resource
		- Added functionalty to initialize new resources with optional variables when given
	- test_irods_unixfilesystem_resource.yml
		- Idempotency changes
		- Test changes

tedgin commented on irods/library/tests/test_irods_unixfilesystem_resource.yml:
```diff
@@ -1,7 +1,88 @@
+    TEST_CONTEXT: context_message_here
+    TEST_STATUS: up
+    TEST_FREE_SPACE: 2000000
+  environment:
+    TEST_PORT: 1247
+    TEST_USER_NAME: rods
+    TEST_HOST: "{{ groups['irods_catalog'][0] }}"
+    TEST_PASSWORD: rods
+    TEST_ZONE_NAME: testing
```
These aren't used, so please remove them.

tedgin commented on irods/library/irods_unixfilesystem_resource:
```diff
@@ -1,195 +1,201 @@
+
+version added: "2.11.11"
+
+description: An ansible module for creating an iRODS Unix filesystem storage
+resource
```
This is incorrect YAML, so the ansible module documentation won't be generated correctly.

tedgin commented on irods/library/irods_unixfilesystem_resource:
```diff
@@ -1,195 +1,201 @@
+description: An ansible module for creating an iRODS Unix filesystem storage
+resource
+
+required variables:
+    name:
+        description:
+            - the name of the resource
+        required: true
+        type: str
+    host:
+        description:
+            - the FQDN or IP address of the resource server hosting this
+            resource
+        required: true
+        type: str
+    vault:
+        description:
+            - the absolute path to the root directory of the vault
+
+optional variables:
+    context:
+        description:
+            - any context to attach to this resource
+        type: str
+    status:
+        description:
+            - starting status 'up' or 'down'
+    init_free_space:
+        description:
+            - amount of free space to allocate
```
This should be one section, called `options`. Not all of the options are documented. Also, some are spelled differently than what's in the code, and some are missing some of the required fields, like `type` and  `required`.

tedgin reviewed (changes requested):
Fix the ansible lint issues and the address the inline comments, please.

Fennersteel committed:
Housekeeping
	- irods_unixfilesystem_resource
		- Edited documentation to be YML-compliant
		- Updated documentation to include all options for input
	- test_irods_unixfilesystem_resource
		- Removed unused variables

tedgin commented on irods/library/irods_unixfilesystem_resource:
```diff
@@ -16,10 +16,10 @@ short_description: Create an iRODS Unix filesystem storage resource
 
-description: An ansible module for creating an iRODS Unix filesystem storage
-resource
+description: 
+    - An ansible module for creating an iRODS Unix filesystem storage resource
```
This comment applies to all `descritption` fields in this section.

When YAML field holds a long string holding text that can be wrapped when rendered, use a `>` followed by a linefeed.  This tells YAML that the next set of indented lines are considered to be a single long string that can wrapped.  Linefeeds can be used to wrap the line to preserve readability in the editor.  For example,

```yaml
field_1: >
  Here is a paragraph that can be
  wrapped.

field_2: ...
```

See the first answer under https://stackoverflow.com/questions/3790454/how-do-i-break-a-string-in-yaml-over-multiple-lines for more information.

tedgin commented on irods/library/irods_unixfilesystem_resource:
```diff
@@ -16,10 +16,10 @@ short_description: Create an iRODS Unix filesystem storage resource
+options:
     name:
         description:
             - the name of the resource
```
If the string fits on the same line as the field, put it on that line. This comment applies to all of the `description` fields in this section.

tedgin reviewed (changes requested):
The ansible-lint issues that I requested be fixed during the last change request weren't addressed. Please address them as well as the new pycodestyle issue before making your next PR.

Fennersteel committed:
Linting changes

tedgin committed:
DS-504: linting WIP

Commit:
DS-504 WIP

tedgin committed:
DS-504: removed unused registers from test
